shared/engine/transport-shared.js

```diff
@@ -24,7 +98,38 @@ class TransportShared extends RobustTransport {
       this.writeCallback = writeCallback
     }
     if (incomingRPCCallback) {
-      this.set_generic_handler(incomingRPCCallback)
+      // delay the call back to us
+      const handler = payload => {
+        this._injectInstrumentedResponse(payload)
+        incomingRPCCallback(payload)
+      }
+
+      this.set_generic_handler(_makeDelayed(_makeLogged(handler, 'serverToEngine'), KEYBASE_RPC_DELAY_RESULT))
+    }
+  }
+
+  // add delay / logging / multiple call checking
+  _injectInstrumentedResponse (payload: any) {
+    if (!payload || !payload.response) {
+      return
+    }
+
+    const oldResponse = payload && payload.response
+
+    if (payload && oldResponse) {
+      const calls = ['cancel', 'error', 'result']
+
+      // Can't use {...} here due to react-native not doing object.assign on objects w/ proto chains
+      payload.response = {}
+      Object.keys(oldResponse).forEach(key => {
+        payload.response[key] = oldResponse[key]
+      })
+
+      calls.forEach(call => {
+        payload.response[call] = _wrap((...args) => {
+          oldResponse[call](...args)
```

Comment:
i think this is a dangerous pattern and should be avoided. the caller changes binding and now you're broken again